packages/client/src/Store/PlayerUi.tsx

```diff
@@ -21,6 +24,21 @@ export const usePlayerUiState = (): PlayerUi => {
   if (context === undefined) {
     throw new Error('usePlayerUiState must be used within a PlayerStateContext.Provider')
   }
+  //process roleid for the legacy data
```

Comment:
I could do with some help here please, @lebaphi 

This happens as we're loading a game. We see if the roleId is missing. If it is missing, we create the roleId for that role, then we push the forces to the database.

Do we need to worry about multiple players updating the roles at the same time, and getting a conflict?

Reply:
Hi @IanMayo , I just wonder if multiple users adding the new role or updating the role at the same time, how do we deal with conflicts?

Reply:
Aah, we don't want to allow multiple admins to create roles at the same time - I _Think_ that's to be treated as a training issue.  The issue is when multiple players load an old wargame at the same time, and separately try to generate the roleId entries.

Reply:
ah okay, so I think we need to handle the conflict error for this thing, no worries 😙

Reply:
I don't know for sure - I'm just thinking it through with you.

Reply:
Let me look at the network traffic & see how long it takes to process with two players logged in.  If it's well under a second, then I don't think the conflict will happen.

Reply:
It _seems to take 0.91ms to run - so I don't think there will be conflicts.

Reply:
yep, it seems unlikely to occur, but if you want to make sure there is no conflict, I think we should handle the conflict error, if it does, just update this document again https://pouchdb.com/guides/conflicts.html

Reply:
Oh yeah - PouchDb will tell us if there is a conflict. Yes, in this instance I think we should just `put` it again.

Reply:
cool, I will update soon 👍🏻

Reply:
Hello @lebaphi - I could do with closing this soon please, since in my code I want to use `roleId` instead of `name`.

If you're not ready to handle the conflict today, then let's merge the PR and solve it as another issue.

Reply:
HI @IanMayo , okay you can merge this PR first, then I can have another PR to resolve the conflict issue later.